user/cp.c

```diff
@@ -1,15 +1,35 @@
+/*
+ * @Date: 2024-09-12
+ * @LastEditors: GoKo-Son626
+ * @LastEditTime: 2024-09-16
+ * @FilePath: /caffeinix/user/cp.c
+ * @Description: 
+ */
 #include "user.h"
 #include "fcntl.h"
+#include "stat.h"
 
 #define TAG                     "cp: "
 
 static int cp(char *src, char *dst)
 {
         int sfd, dfd, n;
         char buf[1024];
+        struct stat st;
+
+        if(stat(src, &st) < 0) {
+                printf(TAG"cannot stat '%s': No such file\n", src);
+                goto r0;
+        }
+
+        if(st.type == T_DIR) {
+                printf(TAG"cannot copy '%s': Is a directory\n", src);
+                goto r0;
+        }
+
         sfd = open(src, O_RDONLY);
         if(!sfd) {
-                printf(TAG"cannot stat '%s': No such file or directory\n", src);
+                printf(TAG"cannot stat '%s': No such file\n", src);
```

Comment:
NO. Please save `or directory`.
Why? Please try to call `cp` command in linux shell and try to copy a file that does not exist.

Reply:
Oh, I know. I'm sorry